bvm/ballerina-core/src/main/java/org/ballerinalang/util/TableProvider.java

```diff
@@ -0,0 +1,310 @@
+/*
+*  Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+*  WSO2 Inc. licenses this file to you under the Apache License,
+*  Version 2.0 (the "License"); you may not use this file except
+*  in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing,
+*  software distributed under the License is distributed on an
+*  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*  KIND, either express or implied.  See the License for the
+*  specific language governing permissions and limitations
+*  under the License.
+*/
+package org.ballerinalang.util;
+
+import org.ballerinalang.model.types.BArrayType;
+import org.ballerinalang.model.types.BStructType;
+import org.ballerinalang.model.types.BTableType;
+import org.ballerinalang.model.types.BType;
+import org.ballerinalang.model.types.TypeTags;
+import org.ballerinalang.model.values.BBlobArray;
+import org.ballerinalang.model.values.BBooleanArray;
+import org.ballerinalang.model.values.BFloatArray;
+import org.ballerinalang.model.values.BIntArray;
+import org.ballerinalang.model.values.BStringArray;
+import org.ballerinalang.model.values.BStruct;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.util.exceptions.BallerinaException;
+
+import java.io.ByteArrayInputStream;
+import java.sql.Blob;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+/**
+ * {@code TableProvider} creates In Memory database for tables.
+ *
+ * @since 0.963.0
+ */
+public class TableProvider {
+
+    private static TableProvider tableProvider = null;
+    private int tableID;
+
+    private TableProvider() {
+        tableID = 0;
+    }
+
+    public static TableProvider getInstance() {
+        if (tableProvider == null) {
+            synchronized (TableProvider.class) {
+                if (tableProvider == null) {
+                    tableProvider = new TableProvider();
+                }
+            }
+        }
+        return tableProvider;
+    }
+
+    private synchronized int getTableID() {
+        return this.tableID++;
+    }
+
+    public String createTable(BType constrainedType) {
+        String tableName = TableConstants.TABLE_PREFIX + ((BTableType) constrainedType).getConstrainedType().getName()
```

Comment:
Can the `constrainedType` be something other than `BTableType`?

Reply:
No. Here [1] it can be only BTableType

[1] https://github.com/anupama-pathirage/ballerina/blob/table-type-support-v1/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolResolver.java#L475